.github/workflows/java-maven-openjdk-codeql.yml

```diff
@@ -0,0 +1,62 @@
+name: "CodeQL"
+
+on:
+  workflow_call:
+    inputs:
+      runs-on:
+        description: |
+          The type of machine to run the job on. Must be provided as a stringified list (e.g. `runs-on: '["ubuntu-latest","self-hosted"]'`)
+        required: true
+        type: string
+      mvn-version:
+        required: false
+        type: number
+        default: 3.9
+      mvn-arguments:
+        required: false
+        type: string
+        default: '--also-make --batch-mode --strict-checksums --update-snapshots -Dmaven.gitcommitid.skip=true'
+      jdk-version:
+        required: true
+        type: number
+
+jobs:
+  analyze-java:
+    name: Analyze Java
+
+    runs-on: ${{ fromJson(inputs.runs-on) }}
+
+    container:
+      image: maven:${{ inputs.mvn-version }}-amazoncorretto-${{ inputs.jdk-version }}-debian:latest
```

Comment:
this is the same container we're proposing to use to build our things. But it shouldn't make a huge difference in the end